src/main/java/com/es/phoneshop/dao/impl/OrderDaoImpl.java

```diff
@@ -0,0 +1,38 @@
+package com.es.phoneshop.dao.impl;
+
+import com.es.phoneshop.dao.OrderDao;
+import com.es.phoneshop.model.order.Order;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class OrderDaoImpl extends GenericDaoImpl<Order> implements OrderDao {
```

Comment:
it's better and more clear when "extends GenericDaoImpl<Order>" is defined at interface level